controller/readwrite.go

```diff
@@ -352,6 +334,97 @@ func (rw *ReadWrite) checkApply(ctx context.Context, d driver.Driver, retry, onc
 	return rendered, nil
 }
 
+func (rw *ReadWrite) createTask(ctx context.Context, taskConfig config.TaskConfig) (driver.Driver, error) {
+	taskConfig.Finalize(rw.conf.BufferPeriod, *rw.conf.WorkingDir)
+	if err := taskConfig.Validate(); err != nil {
+		rw.logger.Trace("invalid config to create task", "error", err)
+		return nil, err
+	}
+
+	taskName := *taskConfig.Name
+	logger := rw.logger.With(taskNameLogKey, taskName)
+
+	// Check if task exists, if it does, do not create again
+	if _, ok := rw.drivers.Get(taskName); ok {
+		logger.Trace("task already exists")
+		return nil, fmt.Errorf("task with name %s already exists", taskName)
+	}
+
+	d, err := rw.createNewTaskDriver(taskConfig)
+	if err != nil {
+		logger.Error("error creating new task driver", "error", err)
+		return nil, fmt.Errorf("error creating new task driver: %v", err)
+	}
+
+	// Initialize the new task
+	err = d.InitTask(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing new task, %s", err)
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+		}
+		ok, err := d.RenderTemplate(ctx)
+		if err != nil {
+			logger.Error("error rendering task template")
+			return nil, fmt.Errorf("error rendering template for task '%s': %s", taskName, err)
+		}
+		if ok {
+			// Once template rendering is finished, break
+			break
+		}
+	}
+
+	// TODO: Set the buffer period
+	// d.SetBufferPeriod()
+
+	// Add the task driver to the driver list
+	err = rw.drivers.Add(taskName, d)
```

Comment:
Only add to drivers on successful create (and run) -- move to TaskCreate and TaskCreateAndRun